core/src/test/scala/org/apache/spark/storage/ShuffleBlockFetcherIteratorSuite.scala

```diff
@@ -254,6 +254,78 @@ class ShuffleBlockFetcherIteratorSuite extends SparkFunSuite with PrivateMethodT
     intercept[FetchFailedException] { iterator.next() }
   }
 
+  test("SPARK-31017: Hit maxBytesInFlight limitation before maxBlocksInFlightPerAddress") {
+    val blockManager = mock(classOf[BlockManager])
+    val localBmId = BlockManagerId("test-client", "test-local-host", 1)
+    doReturn(localBmId).when(blockManager).blockManagerId
+
+    val remoteBmId1 = BlockManagerId("test-remote-client-1", "test-remote-host1", 1)
+    val remoteBmId2 = BlockManagerId("test-remote-client-2", "test-remote-host2", 2)
+    val remoteBmId3 = BlockManagerId("test-remote-client-3", "test-remote-host2", 3)
+    // set maxBlocksInFlightPerAddress to Int.MaxValue,
```

Comment:
seems we don't need this line, as we have it in https://github.com/apache/spark/pull/27767/files#diff-d9abbfedd9c2b0aed76c94c91ec0cec2R283